homeworks/week5/VuNguyen.java

```diff
@@ -0,0 +1,55 @@
+package homeworks.week5;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class VuNguyen {
+    public static void main(String[] args) {
+        int[] boss = {1, 3, 3, -1, -1, 4, 2, 8, 3};
+        System.out.println(lowestCommonBoss(boss, 1, 1));  // 1
+        System.out.println(lowestCommonBoss(boss, 2, 0));  // 3
+        System.out.println(lowestCommonBoss(boss, 6, 0));  // 3
+        System.out.println(lowestCommonBoss(boss, 7, 1));  // 3
+        System.out.println(lowestCommonBoss(boss, 4, 5));  // 4
+        System.out.println(lowestCommonBoss(boss, 6, 5));  // -1
+        System.out.println(lowestCommonBoss(boss, 4, 2));  // -1
+        System.out.println(lowestCommonBoss(boss, 4, 3));  // -1
+        System.out.println(lowestCommonBoss(boss, 4, -1));  // -1
+
+        System.out.println(lowestCommonBoss(boss, 4, -2));  // excep
+        System.out.println(lowestCommonBoss(boss, 5, -1));  // excep
+    }
+
+    // time complexity: worst is O(h)
```

Comment:
in worse case the time complexity is still O(n) as we still need to travel all nodes.